android/expoview/src/main/java/host/exp/exponent/notifications/delegates/ScopedExpoPresentationDelegate.kt

```diff
@@ -41,4 +43,19 @@ class ScopedExpoPresentationDelegate(context: Context) : ExpoPresentationDelegat
     }
     return super.getNotification(statusBarNotification)
   }
+
+  /**
+   * Prevent notifications of different experiences overwriting each other in Expo Go.
+   * The notification is either a remote notification sent through Expo's push
+   * service, or a local notification.
+   */
+  override fun getNotifyId(request: NotificationRequest): Int {
+    var experienceId: String? = "";
+    if (request.trigger is FirebaseNotificationTrigger) {
+      experienceId = (request.trigger as FirebaseNotificationTrigger).remoteMessage.data["experienceId"]
+    } else if (request is ScopedNotificationRequest) {
+      experienceId = request.experienceIdString
+    }
```

Comment:
```suggestion
    val experienceId = if (request.trigger is FirebaseNotificationTrigger) {
      (request.trigger as FirebaseNotificationTrigger).remoteMessage.data["experienceId"]
    } else if (request is ScopedNotificationRequest) {
      request.experienceIdString
    } else {
      ""
    }
```

I think the `request.trigger`  should be automatically cast here. So we don't have to cast it explicitly 🤔

Reply:
Android studio gives an error if we rely on smart cast here, I think bc we are checking `<object>.<property> is xxxx` instead of just `object is xxxx`. To smart cast, would need to do something like 
```
val trigger = request.trigger
if (trigger is FirebaseNotificationTrigger) {trigger.remoteMessage...
```

Reply:
that makes sense 👍